pkg/storages/memory/mock/folder.go

```diff
@@ -75,6 +76,13 @@ func (f *Folder) PutObject(name string, content io.Reader) error {
 	return f.MemFolder.PutObject(name, content)
 }
 
+func (f *Folder) PutObjectWithContext(ctx context.Context, name string, content io.Reader) error {
+	if err := ctx.Err(); err != nil {
```

Comment:
Why not reuse the `contextio.NewReader` here?

Reply:
It is a performance micro optimisation. Memory folder works fast. So I decided to skip context checking on every Read call. Checking context on start should be enough to detect most of cancelling cases.

Reply:
> It is a performance micro optimisation. Memory folder works fast. So I decided to skip context checking on every Read call. Checking context on start should be enough to detect most of cancelling cases.

Since it is being used only for tests, I suggest sticking with the approach that is being used on other storages to catch possible bugs in the future.